src/main/java/org/opensrp/service/EventService.java

```diff
@@ -150,6 +162,13 @@ public synchronized Event addEvent(Event event) {
 		
 		event.setDateCreated(DateTime.now());
 		allEvents.add(event);
+		String planIdentifier = event.getDetails() != null ? event.getDetails().get("planIdentifier") : null;
+		if (isPlanEvaluationEnabled && planIdentifier != null) {
+			PlanDefinition plan = planRepository.get(planIdentifier);
+			if(plan.getStatus().equals(PlanDefinition.PlanStatus.ACTIVE) && (plan.getEffectivePeriod().getEnd() == null ||
+					plan.getEffectivePeriod().getEnd().isAfter(new DateTime().toLocalDate())))
+			taskGenerator.processPlanEvaluation(plan, null, username);
```

Comment:
Pass in the event to  for plan evaluation

Reply:
Done